nvme.c

```diff
@@ -3075,11 +3084,58 @@ static int create_ns(int argc, char **argv, struct command *cmd, struct plugin *
 		return -EINVAL;
 	}
 
-	err = parse_lba_num_si(dev, "nsze", cfg.nsze_si, cfg.flbas, &cfg.nsze);
+	id = nvme_alloc(sizeof(*id));
+	if (!id)
+		return -ENOMEM;
```

Comment:
Just FYI: It may needed to output any error message but not just the error return. This is not only for the changes but exsiting other codes using `nvme_alloc()` also same. So seems better the error message output by the function itself.

Reply:
Testing small allocation fails is pretty futile in userspace, because you will almost certainty never see it. And thus most of the error paths will not work because no one really tests this.

Libraries like GLib just call `exit` on small allocation fails. For bigger allocations it might makes sense to test if it fails and handle it. 

Even worse is that if the small allocation fails printing any error message is also going to fail because the printing path is likely to use memory allocation as well. 

If we really want to improve the situation I'd say we make `nvme_alloc` fail hard on memory allocation failures and remove all the untested `-ENOMEM` code paths.